ui/app/pages/settings/networks-tab/network-form/network-form.component.js

```diff
@@ -211,16 +212,27 @@ export default class NetworkForm extends PureComponent {
     )
   }
 
-  validateUrl = (url, stateKey) => {
-    const invalidUrlErrorMsg = stateKey === 'rpcUrl' ? 'invalidRPC' : 'invalidBlockExplorerURL'
-
-    if (validUrl.isWebUri(url) || (stateKey === 'blockExplorerUrl' && url === '')) {
+  validateRpcUrl = (url, stateKey) => {
+    const { rpcUrls } = this.props
+    if (validUrl.isWebUri(url) || url === '') {
       this.setErrorTo(stateKey, '')
     } else {
       const appendedRpc = `http://${url}`
       const validWhenAppended = validUrl.isWebUri(appendedRpc) && !url.match(/^https?:\/\/$/)
 
-      this.setErrorTo(stateKey, this.context.t(validWhenAppended ? 'uriErrorMsg' : invalidUrlErrorMsg))
+      if (validWhenAppended) this.setErrorTo(stateKey, this.context.t('urlErrorMsg'))
+    }
+
+    if (rpcUrls.includes(url)) {
```

Comment:
We should try to avoid setting the error to `''` then setting it again to something non-blank. Perhaps we can re-order things to check for the two bad cases first, then blank the error in the `else` case? Instead of checking for the blank case first.